cogs/weightcog.py

```diff
@@ -557,6 +560,45 @@ async def export(self, ctx: commands.Context, user: discord.Member = None) -> No
                 file=discord.File(csv_file, filename=f"{user.display_name}_weight_data.csv"),
             )
 
+    @commands.command()
+    async def del_all_log(self, ctx: commands.Context, user: discord.Member = None) -> None:
+        """
+        Deletes all log data (CSV file) associated with a specified Discord user or the command invoker.
+
+        Parameters:
+            ctx (commands.Context): The context of the command.
+            user (discord.Member, optional): The Discord user for whom to delete the log data. Defaults to the command invoker.
+
+        Returns:
+            None: This function does not return any value.
+
+        Raises:
+            discord.ext.commands.MissingPermissions: If the command invoker lacks the necessary permissions.
+            discord.ext.commands.MissingRole: If the command invoker lacks the required bot-input role when specifying another user.
+
+        Example:
+            .del_all_log
+            .del_all_log user(Optional)
+        """
+        if user and (user != ctx.author) and not has_bot_input_perms(ctx):
+            await ctx.send(
+                "You don't have the bot-input role and are therefore not allowed to specify other"
+                " users."
+            )
+            return
+
+        user = user or ctx.author
+        user_id = str(user.id)
+        file_path = os.path.join(self.data_folder, f"{user_id}.csv")
+
+        # if the file/data for user does not exist
+        if not os.path.exists(file_path):
+            await ctx.send("No weight data found for this user.")
+            return
+        else:
+            os.remove(file_path)
```

Comment:
I think its important to prompt the user to ask to delete the data. I would suggest the following:

1. Let the bot respond to the command the user provides.
2. This comment says "Are you sure you want to remove all user weight data for @user"?
3. The bot replies to its own comment with a thumbs up and a thumbs down.
4. If the user presses up the data gets deleted and the bot prompts that its about to delete the user data
5. If the user presses down, or the bot times out, the data does not get deleted and the function returns

Reply:
Implemented, let me know if you would like any prompt or something else to be changed.

<img width="903" alt="Screenshot 2023-10-06 at 9 43 54 PM" src="https://github.com/CodeWizardClemens/BSF-bot/assets/45039724/d6a104b7-1740-4d52-ad8d-d77c74bc8be1">